ballerina-tests/tests/http_client_data_binding_anydata.bal

```diff
@@ -545,3 +588,128 @@ function testXmlArrDatabindingByType() {
+    string|xml person = check clientDBBackendClient->get("/anydataTest/getFormData");
+    if person is string {
+        test:assertEquals(person, "first%20Name=WS%20O2&tea%24%2Am=Bal%40Dance", msg = "Found unexpected output");
+    } else {
+        test:assertFail(msg = "Found unexpected output type");
+    }
+}
+
+@test:Config {}
+function testUnionOfDatabindingForByteArrWithByteArrXml() returns error? {
+    byte[]|xml response = check clientDBBackendClient->get("/anydataTest/byteArrType");
+    if response is byte[] {
+        test:assertEquals(check strings:fromBytes(response), "WSO2", msg = "Found unexpected output");
+    } else {
+        test:assertFail(msg = "Found unexpected output type");
+    }
+}
+
+@test:Config {}
+function testUnionOfDatabindingForRecordWithPersonNStock() returns error? {
+    ClientAnydataDBPerson|ClientAnydataDBStock response = check clientDBBackendClient->get("/anydataTest/recordType");
+    if response is ClientAnydataDBPerson {
+        test:assertEquals(response, {name:"hello", age:20}, msg = "Found unexpected output");
+    } else {
+        test:assertFail(msg = "Found unexpected output type");
+    }
+}
+
+@test:Config {}
+function testNilableUnionOfDatabindingForXml() returns error? {
+    xml|string? response = check clientDBBackendClient->get("/anydataTest/getNoContentXml");
+    test:assertTrue(response is (), msg = "Found unexpected output");
+}
+
+@test:Config {}
+function testNilableUnionOfDatabindingForString() returns error? {
+    xml|string? response = check clientDBBackendClient->get("/anydataTest/getNoContentString");
+    test:assertTrue(response is (), msg = "Found unexpected output");
+}
+
+@test:Config {}
+function testNilableUnionOfDatabindingForByteArr() returns error? {
+    xml|byte[]? response = check clientDBBackendClient->get("/anydataTest/getNoContentString");
+    test:assertTrue(response is (), msg = "Found unexpected output");
+}
+
+@test:Config {}
+function testNilableUnionOfDatabindingForMapFormData() returns error? {
+    xml|map<string>? response = check clientDBBackendClient->get("/anydataTest/getNoContentUrlEncoded");
+    test:assertTrue(response is (), msg = "Found unexpected output");
+}
+
+@test:Config {}
+function testNilableUnionOfDatabindingForStringFormData() returns error? {
+    xml|string? response = check clientDBBackendClient->get("/anydataTest/getNoContentUrlEncoded");
+    test:assertTrue(response is (), msg = "Found unexpected output");
+}
+
+@test:Config {}
+function testNilableUnionOfDatabindingForOctetBlob() returns error? {
+    xml|byte[]? response = check clientDBBackendClient->get("/anydataTest/getNoContentBlob");
+    test:assertTrue(response is (), msg = "Found unexpected output");
+}
+
+@test:Config {}
+function testNilableUnionOfDatabindingForJson() returns error? {
+    xml|json? response = check clientDBBackendClient->get("/anydataTest/getNoContentJson");
+    test:assertTrue(response is (), msg = "Found unexpected output");
+}
```

Comment:
Do we have a test-case to cover union types more than two ?

Reply:
Added via 9a47153f00311d6798ec1db66fbdfbb31f051e66